frontend/src/metabase/visualizations/echarts/cartesian/option/series.ts

```diff
@@ -74,16 +74,45 @@ export const getBarLabelLayout =
 
 export function getDataLabelFormatter(
   seriesModel: SeriesModel,
+  dataset: ChartDataset,
```

Comment:
Added the dataset because we need to pre-check the formatted value lengths to determine if we should use `compact` or `full` formatting